src/ui/dialogs/exstatus.ts

```diff
@@ -0,0 +1,68 @@
+import Dialog from '../Dialog'
+import Client from '../../Client'
+//import JID from '../../JID'
+
+let exstatusTemplate = require('../../../template/exstatus.hbs');
+
+let dialog: Dialog;
+
+export default function() {
+
+   let content = exstatusTemplate();
+
+   dialog = new Dialog(content);
+   let dom = dialog.open();
+
+   $(dom).ready(function(){
+    getStatus();
+   });
+
+   dom.find('form').submit(onSubmit);
+
+   $('.jsxc-js-clear').on('click',removeStatus);
+}
+
+function onSubmit()
+{
+   let account = Client.getAccountManager().getAccounts()[0];
+   let statustext = dialog.getDom().find('#extended_status_text').val().toString();
+   let item = $build('status',{xmlns:'http://jabber.org/protocol/status'}).c('text').t(statustext).tree();
+
+   account.getContact().setStatus(statustext);
+
+   account.getConnection().getPEPService().publish('http://jabber.org/protocol/status',item,'current').then(function(result) {
```

Comment:
In which XEP is this defined?

Reply:
oh i now see it is not official... so we could remove it or not... up to you...
found it here: http://www.funkroom.net/xep/xep-status.html